cmd/erasure-zones.go

```diff
@@ -910,6 +911,115 @@ func leastEntryZone(zoneEntryChs [][]FileInfoCh, zoneEntries [][]FileInfo, zoneE
 	return lentry, leastEntryCount, zoneIndex, isTruncated
 }
 
+// Calculate least entry across zones and across multiple FileInfoVersions
+// channels, returns the least common entry and the total number of times
+// we found this entry. Additionally also returns a boolean
+// to indicate if the caller needs to call this function
+// again to list the next entry. It is callers responsibility
+// if the caller wishes to list N entries to call leastEntry
+// N times until this boolean is 'false'.
+func leastEntryZoneVersions(zoneEntryChs [][]FileInfoVersionsCh, zoneEntries [][]FileInfoVersions, zoneEntriesValid [][]bool) (FileInfoVersions, int, int, bool) {
+	for i, entryChs := range zoneEntryChs {
+		for j := range entryChs {
+			zoneEntries[i][j], zoneEntriesValid[i][j] = entryChs[j].Pop()
```

Comment:
We cannot for last marker we need to remember the last entry which was popped from the channel, we need to keep last entry in the memory, channels have no way to push back into the channel - so this style of Push/Pop style to make sure we have entries remembered from each of the entryCh

So that the merge sort also works neatly.